src/Raven.Server/Indexing/VoronIndexOutput.cs

```diff
@@ -73,11 +114,54 @@ protected override void Dispose(bool disposing)
                     files.AddStream(nameSlice, _file);
                 }
             }
-            finally
+            catch (Exception e)
+            {
+                if (Logger.IsInfoEnabled)
+                    Logger.Info($"Failed to copy the file: {_name}", e);
+
+                _indexOutputFilesSummary.SetWriteError();
+
+                if (e is IOException ioe && IsOutOfDiskSpaceException(ioe))
+                {
+                    // can happen when trying to copy from the file stream
+                    ThrowDiskFullException();
+                }
+                
+                throw;
+            }
+        }
+
+        private void DisposeFile()
+        {
+            try
             {
                 _file.Dispose();
+            }
+            catch (Exception e)
+            {
+                // we are done with this file, nothing we can do here
+                if (Logger.IsInfoEnabled)
+                    Logger.Info($"Failed to dispose the file: {_name} in path: {_fileTempPath}", e);
+            }
+            finally
+            {
                 PosixFile.DeleteOnClose(_fileTempPath);
             }
         }
+
+        private void ThrowDiskFullException()
+        {
+            var directory = Path.GetDirectoryName(_fileTempPath);
+            var driveInfo = DiskSpaceChecker.GetDiskSpaceInfo(directory);
+            throw new DiskFullException($"There isn't enough space to flush the buffer of the file: {_fileTempPath}. " +
+                                        $"Currently available space: {driveInfo?.TotalFreeSpace}");
```

Comment:
Handle the case where this is null and say `space: N/A` or something

Reply:
done